docs/guides/checkpointing_solutions/Emergency_Checkpointing.md

```diff
@@ -0,0 +1,181 @@
+
+* **Persistent Checkpoints**: These are standard checkpoints saved periodically to durable storage(i.e. GCS bucket). They ensure that you can recover your training state even after a complete job failure. This is controlled by `enable_checkpointing`, and `checkpoint_period`.
+
+
+#### `enable_checkpointing`
+A master switch to enable (`True`) or disable (`False`) saving checkpoints during the training run.
+* **Type**: `boolean`
+* **Default**: `False`
+
+#### `enable_emergency_checkpoint`
+When set to (`True`), this flag enables the two-tiered emergency checkpointing feature. 
+* **Type**: `boolean`
+* **Default**: `False`
+
+#### `async_checkpointing`
+ When set to (`True`), this flag makes checkpoint saving
+    asynchronous. The training step is only blocked for the minimal time needed
+    to capture the model's state, and the actual writing to storage happens in a
+    background thread. This is highly recommended for performance. It's enabled
+    by default.
+* **Type**: `boolean`
+* **Default**: `True`
+
+#### `local_checkpoint_directory`
+The high-speed local filesystem path(i.e. ramdisk) where **emergency checkpoints** are saved. Setting this path, along with a non-zero `local_checkpoint_period`, enables the emergency checkpointing feature.
+* **Type**: `string`
+* **Default**: `""`
+
+#### `local_checkpoint_period`
+The interval, in training steps, for how often a local **emergency checkpoint** is saved. This should be set to a much smaller value than `checkpoint_period` for frequent, low-overhead saves.
+* **Type**: `integer`
+* **Default**: `0`
+
+#### `checkpoint_period`
+The interval, in training steps, for how often a checkpoint is saved to **persistent storage**.
+* **Type**: `integer`
+* **Default**: `10000`
+
+#### `enable_single_replica_ckpt_restoring`
+If `True`, one replica reads the checkpoint from storage and then broadcasts it to all other replicas. This can significantly speed up restoration on multi-host systems by reducing redundant reads from storage.
+* **Type**: `boolean`
+* **Default**: `False`
+
+---
+## Workload Creation using XPK
+
+Flags below would give the user access to the Ramdisk in their workload:
+
+* `--mtc-enabled`: Enables the Multi-Tier Checkpointing feature, by mounting ramdisk to the workload pods, using csi drivers.
+* `--ramdisk-directory`: Specifies the mount path inside each pod where the high-speed ramdisk will be accessible. Your training application should write its local, emergency checkpoints to this path.
+
+### Example XPK workload creation command
+
+```
```

Comment:
Same as above, my suggestion is to break these code blocks to allow for easy copy-pasting.